bastion/src/path.rs

```diff
@@ -0,0 +1,459 @@
+use crate::context::{BastionId, NIL_ID};
+use std::fmt;
+use std::result::Result;
+
+#[derive(Clone)]
```

Comment:
Doc comment for this since it's public + examples.

Reply:
Still missing an example ;)